packages/enhanced/src/lib/container/HoistContainerReferencesPlugin.ts

```diff
@@ -4,27 +4,123 @@ import type {
   Chunk,
   WebpackPluginInstance,
 } from 'webpack';
+import { normalizeWebpackPath } from '@module-federation/sdk/normalize-webpack-path';
 import ContainerEntryModule from './ContainerEntryModule';
 
+const runtime = require(
+  normalizeWebpackPath('webpack/lib/util/runtime'),
+) as typeof import('webpack/lib/util/runtime');
+
 /**
  * This class is used to hoist container references in the code.
  * @constructor
  */
-export class HoistContainerReferences implements WebpackPluginInstance {
+export class HoistContainerReferencesPlugin implements WebpackPluginInstance {
+  integrateChunks(
+    chunkA: Chunk,
+    chunkB: Chunk,
+    compilation: Compilation,
+  ): void {
+    const { chunkGraph, compiler } = compilation;
+    // do not sort chunk by smallest name, this will cause non-deterministic chunk integration
+
+    // if (chunkA.name && chunkB.name) {
```

Comment:
This is useful but causes issues, integration of chunk-by-byte size is not reliable for tactical chunk integrations. 

These chunks typically cannot be integrated, and to do so requires some special handling of conditions to perform safe chunk merging with runtime and entry points in a non-destructive manner.